core/src/main/java/io/confluent/rest/Application.java

```diff
@@ -103,7 +105,7 @@
 
   protected ApplicationServer<?> server;
   protected Metrics metrics;
-  protected final CustomRequestLog requestLog;
+  protected RequestLog requestLog;
```

Comment:
I would prefer this field is final then the constructor with `RequestLog` public instead (in previous commit), was there a problem with that approach?

Reply:
https://github.com/confluentinc/kafka-rest/pull/1168/files#diff-0dbc682fe5c6635c3e6dc268c10b1adc54544e47ddd2e86815175935fcba24e7R143

`KafkaRestApplication` won't be able to set `requestLog` either from `KafkaRestApplication` ctor, or configureHandler().

Reply:
You could do this with static method though, for example
```
  public KafkaRestApplication(KafkaRestConfig config, String path, String listenerName) {
    super(config, path, listenerName, createRequestLog(config, CustomRequestLog.EXTENDED_NCSA_FORMAT + " %{ms}T"));

    restResourceExtensions =
        config.getConfiguredInstances(
            KafkaRestConfig.KAFKA_REST_RESOURCE_EXTENSION_CONFIG, RestResourceExtension.class);
    config.setMetrics(metrics);
  }
  
  private static RequestLog createRequestLog(KafkaRestConfig config, String requestLogFormat) {
    Slf4jRequestLogWriter logWriter = new Slf4jRequestLogWriter();
    logWriter.setLoggerName(config.getString(RestConfig.REQUEST_LOGGER_NAME_CONFIG));
    // %{ms}T logs request time in milliseconds
    return new CustomRequestLog(logWriter, requestLogFormat);
  }
```